tests/event_queue.cpp

```diff
@@ -0,0 +1,164 @@
+#include <chrono>
+#include <cstdlib>
+#include <deque>
+#include <iostream>
+#include <map>
+#include <string>
+
+#include "doctest.h"
+
+#include "nlohmann/json.hpp"
+#include "test_utils.hpp"
+using namespace test_utils;
+
+using namespace cly;
+using namespace std::literals::chrono_literals;
+
+TEST_CASE("Tests that use the default value of event queue threshold ") {
+  clearSDK();
+  Countly &countly = Countly::getInstance();
+  test_utils::initCountlyWithFakeNetworking(true, countly);
+
+  SUBCASE("Adding events over the threshold size should trigger the events to be sent to the RQ") {
+    // generate 120 events
+    test_utils::generateEvents(120, countly);
+
+    // default threshold is 100 so we should have 20 events in the EQ left
+    CHECK(countly.checkPersistentEQSize() == 20);
+
+    // RQ should have the 100 events
+    test_utils::checkTopRequestEventSize(100, countly);
+  }
+
+  SUBCASE("Adding events 'at' the threshold should trigger the events to be sent to the RQ") {
+    // generate 99 events
+    test_utils::generateEvents(99, countly);
+
+    // default threshold is 100 so we should have 99 events in the EQ still
+    CHECK(countly.checkPersistentEQSize() == 99);
+
+    countly.processRQDebug();
+    // local HTTP request queue should be empty => no events sent to RQ
+    CHECK(http_call_queue.empty());
+
+    // add one more event
+    cly::Event event("click", 2);
+    countly.addEvent(event);
+
+    // reached threshold so we should have 0 events in the EQ
+    CHECK(countly.checkPersistentEQSize() == 0);
+
+    // RQ should have the 100 events
+    test_utils::checkTopRequestEventSize(100, countly);
+  }
+}
+
+TEST_CASE("Tests that use a custom value of event queue threshold") {
+  clearSDK();
+  Countly &countly = Countly::getInstance();
+
+  SUBCASE("Custom threshold size should be used instead of the default one") {
+    countly.setEventsToRQThreshold(90); // before start
+    test_utils::initCountlyWithFakeNetworking(true, countly);
+
+    // generate 100 events
+    test_utils::generateEvents(100, countly);
+
+    // new threshold size is 90 so we should have 10 events in the EQ left
+    CHECK(countly.checkPersistentEQSize() == 10);
+
+    // RQ should have the 90 events
+    test_utils::checkTopRequestEventSize(90, countly);
+  }
+
+  // Setting the threshold size after start
+  SUBCASE("Custom threshold size should be used instead of the default one") {
+    test_utils::initCountlyWithFakeNetworking(true, countly);
+    countly.setEventsToRQThreshold(90); // after start
+
+    // generate 100 events
+    test_utils::generateEvents(100, countly);
+
+    // new threshold size is 90 so we should have 10 events in the EQ left
+    CHECK(countly.checkPersistentEQSize() == 10);
+
+    // RQ should have the 90 events
+    test_utils::checkTopRequestEventSize(90, countly);
+  }
+
+  // Setting a negative threshold size
+  SUBCASE("UINT_MAX should be used instead of the default one") {
```

Comment:
tests name

Reply:
this test just needs to be reskinned 
add also a second call after init and validate again